### PR TITLE
Fix ambiguity around ROUTE_ADVERTISEMENT

### LINKED ISSUE #136: Proxy capsule handling requirements
From @martinduke:

> (S6) What if a capsule IP address range includes a link-local address? It's not a protocol violation  since /0 is legal. Does the receiver simply omit link-local addresses from a broader range when picking an address?
> 
> (S6) It's interesting to say that the sender MUST reset the stream if the PMTU isn't big enough for DATAGRAM. I vaguely recall an extended discussion on this, but a MUST sets up weird incentives to use Capsules instead of datagram. (No action item on this bullet, I think).

### PR DESCRIPTION
**Pull request by DavidSchinazi**
Fixes #136

### CODE REVIEW

File: draft-ietf-masque-connect-ip.md [hunk 1]
```diff
@@ -693,8 +693,8 @@ If any of the capsule fields are malformed upon reception, the receiver of the
 capsule MUST follow the error handling procedure defined in
 {{Section 3.3 of HTTP-DGRAM}}.
 
-Upon receiving the ROUTE_ADVERTISEMENT capsule, an endpoint MAY start routing IP
-packets in these ranges to its peer.
+Upon receiving the ROUTE_ADVERTISEMENT capsule, an endpoint MAY install
+corresponding entries into its routing table.
```
Comment (tfpauly) [inline draft-ietf-masque-connect-ip.md:697]: In many cases, won't this be equivalent? If I don't have a route to send on, I generally won't be able to send.

I also don't love changing the text to imply that a routing table is used by an implementation to provide access to the CONNECT-IP forwarding session. That's sometimes the case, but there are other models (like just having a flow-scoped connection). Elsewhere, we mention routing tables only as examples, not as a normative behavior.

**Review by tfpauly — CHANGES_REQUESTED** [review]
I think for some cases this ends up being equivalent — if I have no routes, I can't send, often.

I also don't love this change implying that

File: draft-ietf-masque-connect-ip.md (hunk 1 above)
Comment (tfpauly) [reply draft-ietf-masque-connect-ip.md:697]:
```suggestion
Upon receiving the ROUTE_ADVERTISEMENT capsule, an endpoint MAY update its local
state about what its peer is willing to route (subject to local policy), such as by installing
entries in a routing table.
```

File: draft-ietf-masque-connect-ip.md [hunk 2]
```diff
@@ -693,8 +693,9 @@ If any of the capsule fields are malformed upon reception, the receiver of the
 capsule MUST follow the error handling procedure defined in
 {{Section 3.3 of HTTP-DGRAM}}.
 
-Upon receiving the ROUTE_ADVERTISEMENT capsule, an endpoint MAY start routing IP
-packets in these ranges to its peer.
+Upon receiving the ROUTE_ADVERTISEMENT capsule, an endpoint MAY update its local
+state about what its peer is willing to route (subject to local policy), such as by installing
+entries in a routing table.
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-ip.md:698]:
```suggestion
state about what its peer is willing to route (subject to local policy), such
as by installing entries in a routing table.
```

File: draft-ietf-masque-connect-ip.md [hunk 3]
```diff
@@ -693,8 +693,9 @@ If any of the capsule fields are malformed upon reception, the receiver of the
 capsule MUST follow the error handling procedure defined in
 {{Section 3.3 of HTTP-DGRAM}}.
 
-Upon receiving the ROUTE_ADVERTISEMENT capsule, an endpoint MAY start routing IP
-packets in these ranges to its peer.
+Upon receiving the ROUTE_ADVERTISEMENT capsule, an endpoint MAY update its local
+state about what its peer is willing to route (subject to local policy), such
```
Comment (DavidSchinazi) [inline draft-ietf-masque-connect-ip.md:697]:
```suggestion
state regarding what its peer is willing to route (subject to local policy), such
```

**Review by tfpauly — APPROVED** [review]

**Review by martinduke — APPROVED** [review]

**Review by gloinul — APPROVED** [review]

**Pull request merged**